File: CSCI361/code/test/prime.py
#!/usr/bin/env python3

def test_prime(test_value):

    if test_value == 2:
        return 1

    for x in range(2, test_value):
        if test_value % x == 0:
            return 0

    return 1

def main():
    test_value = 179424673

    if test_prime(test_value):
        print(test_value, "is prime.")
    else:
        print(test_value, "is not prime.")

if __name__ == '__main__':
    main()

File: CSCI361/code/test/prime.c
#include <stdlib.h>
#include <stdio.h>

int check_prime(int test_value) {

    if (test_value < 2) return 0;
    if (test_value == 2) return 1;

    int i;
    for (i = 2; i < test_value; i++)
        if ((test_value % i) == 0) return 0;

    return 1;
}

int main() {

    int test_value = 179424673;

    if (check_prime(test_value)) printf("%i is prime.\n", test_value);
    else printf("%i is not prime.\n", test_value);

    return 0;
}